src/metrics.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+var util = require('./util');
+var moment = require('moment');
+var eventObj = require('./core/event/event');
+
+class Metrics {
+    constructor(eventBus) {
+        var self = this;
+        self._eventBus = eventBus;
+
+        self._eventBus.addEventListener(eventObj.UNKNOWN_USER_INPUT, this, function(param) {
+            var emitParams = {
+                type: 'errorReport',
+                subtype: 'default',
+                doc: {
+                    input: param.input,
+                    date: param.date
+                },
+                _cb: param._cb
+            }
+            self._eventBus.emitEvent(eventObj.DATABASE_ADD, emitParams);
+        });
+
+        self._eventBus.addEventListener(eventObj.COMMAND_REPORT_ADD, this, function(param) {
+            var inputIndividualData = {
+                date: moment().format('x'),
+                input: param.inputParams
+            }
+            var emitParams = {
+                type: 'commandReport',
+                fetchParams: {
```

Comment:
Sorry for the typo on the google doc, this should be called 'params' like the other events